src/charm.py

```diff
@@ -206,34 +205,34 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
         """Handle collect unit status event (called after every event)."""
         if not self.model.relations.get("credentials"):
             event.add_status(BlockedStatus("Keystone is not related"))
+            return
 
         if not self._get_keystone_data():
             event.add_status(WaitingStatus("Waiting for credentials from keystone"))
+            return
 
         if not self.model.relations.get("cos-agent"):
             event.add_status(BlockedStatus("Grafana Agent is not related"))
-
-        upstream_snap = get_installed_snap_service(UPSTREAM_SNAP)
+            return
```

Comment:
Interesting, search all `event.add_status` in canonical orgs some are using `event.add_status` follow by `return` some don't. I don't quite get the idea of `return` after `event.add_status`, it makes the priority feature useless.

Reply:
Priority feature is useful, however here we want to shortcircuit the logic.  Otherwise we need to re-check the conditions above before checking for snap_service.is_active()

Reply:
https://github.com/canonical/sdcore-upf-operator/blob/b04abd2e64a17daabc043b5e2b131a889ddeeb87/src/charm.py#L72 vs
https://github.com/canonical/identity-platform-admin-ui-operator/blob/d109f04511c171b5c4f65bb98d170da569e7039d/src/charm.py#L321

Reply:
I prefer short-circuiting; the other approach smells full of potential and unexpected bug.